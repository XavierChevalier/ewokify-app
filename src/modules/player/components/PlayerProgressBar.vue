<script setup lang="ts">
  import { computed } from 'vue'
  import { isPropertyValid } from '@/tools/component-properties/PropertyValidator'
  import { number } from 'yup'

  const props = defineProps({
    progress: {
      type: Number,
      required: true,
      // eslint-disable-next-line vue/valid-define-props
      validator: isPropertyValid(number().integer().min(0)),
    },
  })

  const progressFill = computed(() => props.progress + '%')
</script>

<template>
  <div class="h-1 bg-primary-light">
    <div
      class="h-full rounded-full bg-primary"
      :style="{ width: progressFill }"
    />
  </div>
</template>
