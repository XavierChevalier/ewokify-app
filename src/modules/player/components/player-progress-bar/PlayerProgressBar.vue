<script setup lang="ts">
  import { z } from 'zod'
  import { computed } from 'vue'

  const props = defineProps({
    progress: {
      type: Number,
      required: true,
      // eslint-disable-next-line vue/valid-define-props
      validator: (value) => z.number().gte(0).safeParse(value).success,
    },
  })

  const progressFill = computed(() => props.progress + '%')
</script>

<template>
  <div class="h-1 bg-primary-light">
    <div
      class="h-full rounded-full bg-primary"
      :style="{ width: progressFill }"
    />
  </div>
</template>
