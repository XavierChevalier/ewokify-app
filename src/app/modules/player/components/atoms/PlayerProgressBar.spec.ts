import { generateStorybookSnapshotTests } from '@/tests/tools/StorybookSnapshot'
import {
  Progress0,
  Progress100,
  Progress42,
  Progress50,
} from './PlayerProgressBar.stories'
import PlayerProgressBar from './PlayerProgressBar.vue'
import { ComponentTestsGenerator } from '@/tests/tests-generators/ComponentTestsGenerator'

describe('PlayerProgressBar', () => {
  generateStorybookSnapshotTests({
    Progress0,
    Progress42,
    Progress50,
    Progress100,
  })

  const componentTests = new ComponentTestsGenerator(PlayerProgressBar)
  componentTests.itShouldBeDefined()

  componentTests
    .property<number>('progress')
    .itShouldBeDefined()
    .itShouldHaveAValidatorFunction()
    .itShouldSuccessWith(0)
    .itShouldSuccessWith(1)
    .itShouldFailWith('negative number', -1)

  componentTests.generateTests()
})
