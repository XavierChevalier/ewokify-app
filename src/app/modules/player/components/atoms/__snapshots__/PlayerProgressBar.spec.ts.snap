// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerProgressBar [snapshot] should render the "Progress0" component correctly 1`] = `
<div
  class="overflow-hidden h-1 rounded-full bg-primary/20"
>
  <div
    class="h-full bg-primary"
    style="width: 0%;"
  />
</div>
`;

exports[`PlayerProgressBar [snapshot] should render the "Progress42" component correctly 1`] = `
<div
  class="overflow-hidden h-1 rounded-full bg-primary/20"
>
  <div
    class="h-full bg-primary"
    style="width: 42%;"
  />
</div>
`;

exports[`PlayerProgressBar [snapshot] should render the "Progress50" component correctly 1`] = `
<div
  class="overflow-hidden h-1 rounded-full bg-primary/20"
>
  <div
    class="h-full bg-primary"
    style="width: 50%;"
  />
</div>
`;

exports[`PlayerProgressBar [snapshot] should render the "Progress100" component correctly 1`] = `
<div
  class="overflow-hidden h-1 rounded-full bg-primary/20"
>
  <div
    class="h-full bg-primary"
    style="width: 100%;"
  />
</div>
`;
